clients/src/test/java/org/apache/kafka/clients/producer/KafkaProducerTest.java

```diff
@@ -172,6 +173,115 @@ public void testOverwriteAcksAndRetriesForIdempotentProducers() {
                 config.getString(ProducerConfig.TRANSACTIONAL_ID_CONFIG)));
     }
 
+    @Test
+    public void testAcksAndIdempotenceForIdempotentProducers() {
+        Properties validProps = new Properties() {{
+            setProperty(
+                ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9999");
+            setProperty(
+                ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
+            setProperty(
+                ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
+            setProperty(
+                ProducerConfig.ACKS_CONFIG, "0");
+            setProperty(
+                ProducerConfig.ENABLE_IDEMPOTENCE_CONFIG, "false");
+        }};
+        ProducerConfig config = new ProducerConfig(validProps);
+        assertFalse(
+            config.getBoolean(ProducerConfig.ENABLE_IDEMPOTENCE_CONFIG),
+            "idempotence should be overwritten");
+        assertEquals(
+            config.getString(ProducerConfig.ACKS_CONFIG),
+            "0",
+            "acks should be overwritten");
+
+        Properties validProps2 = new Properties() {{
+            setProperty(
+                ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, "localhost:9999");
+            setProperty(
+                ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
+            setProperty(
+                ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, StringSerializer.class.getName());
+            setProperty(
+                ProducerConfig.TRANSACTIONAL_ID_CONFIG, "transactionalId");
+        }};
+        assertDoesNotThrow(
+            () -> new ProducerConfig(validProps2),
+            "Idempotence should be enable by default, " +
+                "ack should be set to all by default, " +
+                "thus specifying a transaction id should be ok");
```

Comment:
We could check ENABLE_IDEMPOTENCE_CONFIG and ACKS_CONFIG instead of `assertDoesNotThrow`?